### ISSUE TITLE
Hazelcast Spring Create Replicated Map 

### ISSUE THREAD

**Issue by FatihErdem** [hazelcast/hazelcast#13120]
Hi,

I  use Spring @Cacheable annotation. With this annotation Hazelcast creates a Map as default. But I want to create Replicated Map for caches which created by @Cacheable annotation.

 I tried to extend `com.hazelcast.spring.cache.HazelcastCacheManager` and override `HazelcastCacheManager#getCache` but no luck. 

How can I achieve this with `hazelcast-spring`?

**Comment by gurbuzali** [comment]
Hazelcast does not provide out of the box configuration for ReplicatedMap but overriding `HazelcastCacheManager#getCache` should have worked. Can you share an example ?

**Comment by FatihErdem** [comment]
Thank you for fast response. 

I changed the  `hazelcastInstance.getMap(name)` to `hazelcastInstance.getReplicatedMap(name)` in `HazelcastCacheManager#getCache` method. But eventually I hit the wall because can not cast `ReplicatedMap` to `com.hazelcast.spring.cache.HazelcastCache`.  And `#getCache` is returns `Cache`.

Is there any example for this?

**Comment by leszko** [comment]
@FatihErdem We don't support ReplicatedMap at the moment. Your contribution on that would be welcome.

**Comment by ahmetmircik** [comment]
@FatihErdem maybe an alternative solution is to enable [near-caching](http://docs.hazelcast.org/docs/latest/manual/html-single/#near-cache) on imap, did you try that?

**Comment by FatihErdem** [comment]
@ahmetmircik Isn't the default configuration already `IMap`? `com.hazelcast.spring.cache.HazelcastCacheManager#getCache` creates a `IMap` instance. But I want to replicate all elements of the map across the cluster. That's why I want to create `ReplicatedMap`.

For now I am using `near-cache` with `IMap`.

**Comment by ahmetmircik** [comment]
no, by default there is no near-cache on imap.

**Comment by FatihErdem** [comment]
@ahmetmircik Yes, I know that. I meant @Cacheable's default is IMap. I already configured near-cache with IMap. I just asked for default ReplicatedMap with @Cacheable which is not possible for now as I understand.
